base/rational.jl

```diff
@@ -129,12 +129,344 @@ Bool(x::Rational) = x==0 ? false : x==1 ? true :
+        if (
+            romo_is_af |
+            (
+                romo_is_to_nearest &
+                (
+                    to_nearest_is_greater_than_half |
+                    (
+                        to_nearest_is_tied &
+                        !mantissa_is_one &
+                        (
+                            romo_is_ntaf |
+                            (
+                                romo_is_ntte &
+                                !mantissa_is_even
+                            )
+                        )
+                    )
+                )
+            )
+        )
+            # Round up
+
+            # Assuming `T` is wide enough and there's no overflow.
+            mantissa += true
+
+            # We need to decrease the bit width in case it increased.
+            mantissa_carry = ispow2(mantissa)
+            mantissa >>= mantissa_carry
+        elseif romo_is_to_nearest & to_nearest_is_tied & mantissa_is_one
+            # Mantissa is one, which means the precision is also
+            # one. Be consistent with the `BigFloat` behavior, for
+            # example: `BigFloat(3) == BigFloat(3.0) == 4`.
+            mantissa_carry = true
+        end
+    end
+
+    # `mantissa` should now again be exactly `requested_bit_width` bits
+    # wide.
+
+    exp = bit_shift_2 - bit_shift_1 + mantissa_carry
+
+    (
+        mantissa = mantissa,
+        exponent = exp,
+        is_exact = result_is_exact,
+    )
+end
+
+# `num`, `den` are positive integers. `requested_bit_width` is the
+# requested floating-point precision. `T` is the integer type that
+# we'll be working with mostly, it needs to be wide enough.
+function rational_to_float_components(
+    num::Integer,
+    den::Integer,
+    requested_bit_width::Integer,
+    ::Type{T},
+    romo::RationalToFloatRoundingMode.R,
+) where {T<:Integer}
+    (false < requested_bit_width) || error("nonpositive bit width")
+
+    # Factor out powers of two
+    trailing_zeros_num = trailing_zeros(num)
+    trailing_zeros_den = trailing_zeros(den)
+    num >>= trailing_zeros_num
+    den >>= trailing_zeros_den
+
+    c = rational_to_float_components_impl(
+        num, den, requested_bit_width, T, romo,
+    )
+
+    (
+        mantissa = c.mantissa,
+        exponent = c.exponent + trailing_zeros_num - trailing_zeros_den,
+        is_exact = c.is_exact,
+    )
+end
+
+function rational_to_float_impl(
+    x::Rational{Bool},
+    ::Type{<:Integer},
+    ::Integer,
+    to_float::C,
+) where {C<:Union{Type,Function}}
+    n = numerator(x)
+    (
+        denominator(x) ?
+        (
+            n ? to_float(true) : to_float(false)
+        ) :
+        (
+            n ? to_float(Inf) : to_float(NaN)
+        )
+    )
+end
+
+# Assuming the wanted rounding mode is round to nearest with ties to
+# even.
+#
+# `requested_precision` must be positive.
+function rational_to_float_impl(
+    x::Rational{S},
+    ::Type{T},
+    requested_precision::Integer,
+    to_float::C,
```

Comment:
What is `to_float`? Is this the target return type..? If so, such arguments are typically put in front here. See the [docs](https://docs.julialang.org/en/v1/manual/style-guide/#Write-functions-with-argument-ordering-similar-to-Julia-Base) for more information on argument ordering in function signatures.